packages/eslint-plugin-blade/src/rules/no-cross-platform-imports.ts

```diff
@@ -0,0 +1,43 @@
+import type { TSESLint } from '@typescript-eslint/utils';
+
+type MessageIds = 'invalidWebImport' | 'invalidNativeImport';
+
+const noCrossPlatformImportsRule: TSESLint.RuleModule<MessageIds> = {
+  defaultOptions: [],
+  meta: {
+    type: 'problem',
+    messages: {
+      invalidWebImport: 'Avoid importing from a .web file in a .native file.',
+      invalidNativeImport: 'Avoid importing from a .native file in a .web file.',
+    },
+    schema: [], // no options
+  },
+  create: (context) => ({
+    ImportDeclaration(node) {
+      const sourceValue = node.source.value;
+      const isWebImport = sourceValue.endsWith('.web');
+      const isNativeImport = sourceValue.endsWith('.native');
+
+      if (
+        isWebImport &&
```

Comment:
I wonder if we have another check for: 
Ideally files which runs on both platform should not import platform specific files

- .native or .web file being imported in file with no extension (files which run on both platforms)

Reply:
>.native or .web file being imported in file with no extension (files which run on both platforms)

What's the issue in importing if the file can run cross-platform?

Reply:
We don't at the moment. Should I add it?

Reply:
Added in https://github.com/razorpay/blade/pull/1495/commits/f24a76c2857e3bee6fab57075d20074316c36667